mne_bids_pipeline/steps/preprocessing/_04_frequency_filter.py

```diff
@@ -355,10 +302,15 @@ def main(*, config: SimpleNamespace) -> None:
                 subject=subject,
                 session=session,
                 run=run,
+                task=task,
             )
             for subject in get_subjects(config)
             for session in get_sessions(config)
-            for run in get_runs(config=config, subject=subject)
+            for run, task in get_runs_tasks(
```

Comment:
The key change is to iterate over `run, task` pairs. So the experimental runs typically end up with `run, cfg.task` and then the noise and rest (if present) end up as `None, "noise"` and `None, "rest"` respectively